Hackaton 2024: resolved some doc issues (#2574)
openturns/openturns: python/doc/theory/data_analysis/kolmogorov_test.rst

.. _kolmogorov_smirnov_test:

> Kolmogorov-Smirnov test

Reviewer: There are 2 pages in the doc called "Kolmogorov-Smirnov test": `kolmogorov_test.rst` and `smirnov_test.rst` depending on whether 2 samples are tested or a sample and a distribution. Does it make sense to have this separation in the doc? Note that this is not done that way in `graphical_fitting_test.rst` where both use cases are presented.

Author: I have seen that point. I guess that we wanted to help the User and lead him straight to the right section very quickly. I have seen that the graphical tests gather all the situations.
I propose to leave it like it is. It is not a problem.

Reviewer: I agree, I was just wondering.

Reviewer 2: I agree with Joseph: there cannot be two pages with the same title. Here is the title given in [^Ross2004] page 504.
```suggestion
The Kolmogorov-Smirnov goodness of fit test for continuous data
```

Please don't forget to extend the underline string below so that it matches the new length.

[^Ross2004]: Ross, S. (2004). Probability and statistics for engineers and scientists. Third Edition. Elsevier.

Reviewer: I guess this is settled then, you may resolve this conversation.

-----------------------

The Kolmogorov-Smirnov test is a statistical test of whether a given sample of data is drawn from a given
probability distribution which is of dimension 1 and continuous.

We denote by :math:`\left\{ x_1,\ldots,x_{\sampleSize} \right\}` the data of dimension 1.
Let :math:`F` and :math:`p` be respectively the cumulative distribution function and the probability
density function of the continuous distribution.

This test involves the calculation of the test statistic which is the weighted maximum
distance between the empirical cumulative distribution function
:math:`\widehat{F}` built from the sample and the tested cumulative distribution function of the continuous

> distribution :math:`F`. The test statistic is defined by:

Reviewer: ```suggestion
distribution :math:`F`. 
Letting :math:`X_1, \ldots , X_\sampleSize` be i.i.d. random variables following the distribution with CDF :math:`F`, the test statistic is defined by:
```

.. math::

>     \widehat{D} = \sqrt{\sampleSize} \sup_x \left|\widehat{F}\left(x\right) - F\left(x\right)\right|

Reviewer: ```suggestion
    \widehat{D} = \sqrt{\sampleSize} \sup_{1 \leq i \leq \sampleSize} \left|\widehat{F}\left(X_i \right) - F\left(X_i \right)\right|
```

The empirical value of the test statistic, evaluated from the sample is:

.. math::

>     \widehat{d} = \sqrt{\sampleSize} \sup_{x_i} \left|\widehat{F}\left(x_i\right) - F\left(x_i\right)\right|

Reviewer: ```suggestion
    \widehat{d} = \sqrt{\sampleSize} \sup_{1 \leq i \leq \sampleSize} \left|\widehat{F}\left(x_i\right) - F\left(x_i\right)\right|
```

The distribution of the test statistic :math:`\widehat{D}` is
known: algorithms are available to compute the distribution of :math:`\widehat{D}` both for :math:`\sampleSize`
large (asymptotic distribution) or for :math:`\sampleSize` small (exact distribution). Then we can use that
distribution to apply the test as follows.
We fix a risk :math:`\alpha` and we evaluate the associated critical value :math:`d_\alpha` which is the quantile of order
:math:`(1-\alpha)` of :math:`\widehat{D}`.

Then a decision is made, either by comparing the test statistic to the theoretical threshold :math:`d_\alpha`
(or equivalently
by evaluating the p-value of the sample  defined as :math:`\Prob{\widehat{D} > \widehat{d}}` and by comparing
it to :math:`\alpha`):

-  if :math:`\widehat{d}>d_{\alpha}` (or equivalently :math:`\Prob{\widehat{D} > \widehat{d}} \leq \alpha`),
   then we reject the continuous distribution with a risk of error :math:`\alpha`,

-  if :math:`\widehat{d} \leq d_{\alpha}` (or equivalently :math:`\Prob{\widehat{D} > \widehat{d}} \geq \alpha`),

>    then the continuous is accepted with a risk of error :math:`\alpha`.

Reviewer: ```suggestion
   then the continuous distribution is considered acceptable.
```

The Kolmogorov-Smirnov goodness of fit test for continuous data
---------------------------------------------------------------

The Kolmogorov-Smirnov test is a statistical test of whether a given sample of data is drawn from a given
probability distribution which is of dimension 1 and continuous.

We denote by :math:`\left\{ x_1,\ldots,x_{\sampleSize} \right\}` the data of dimension 1.
Let :math:`F` be  the (unknown) cumulative distribution function of the continuous distribution.

We want to test  whether the sample is drawn from the cumulative distribution function :math:`G`.

This test involves the calculation of the test statistic which is the weighted maximum
distance between the empirical cumulative distribution function
:math:`\widehat{F}` built from the sample and :math:`G`.
Letting :math:`X_1, \ldots , X_\sampleSize` be i.i.d. random variables following
the distribution with CDF :math:`F`, the test statistic is defined by:

.. math::

    D_{\sampleSize} = \sqrt{\sampleSize} \sup_{1 \leq i \leq \sampleSize} \left|\widehat{F}\left(X_i \right) - G\left(X_i \right)\right|

The empirical value of the test statistic, evaluated from the sample is:

.. math::

>     d_{\sampleSize} = \sqrt{\sampleSize} \sup_{1 \leq i \leq \sampleSize} \left|\widehat{F}\left(x_i\right) - G\left(x_i\right)\right|

Reviewer: Same here: the sup is on i, not x_i

Author: I removed the sum and let the notation sup_x.

.. _kolmogorov_smirnov_test:

The Kolmogorov-Smirnov goodness of fit test for continuous data
---------------------------------------------------------------

The Kolmogorov-Smirnov test is a statistical test of whether a given sample of data is drawn from a given
probability distribution which is of dimension 1 and continuous.

Let :math:`\left\{ x_1,\ldots, x_{\sampleSize} \right\}` be a sample of dimension 1 drawn from the (unknown) cumulative distribution function :math:`F`.

We want to test  whether the sample is drawn from the cumulative distribution function
:math:`G`.

This test involves the calculation of the test statistic which is the weighted maximum
distance between the empirical cumulative distribution function
:math:`F_{\sampleSize}` and :math:`G`.
Letting :math:`X_1, \ldots , X_{\sampleSize}`  be independent random variables respectively distributed according to :math:`F`, then :math:`F_{\sampleSize}` is defined by:

.. math::

    F_{\sampleSize}(x) & = \sum_{i=1}^{\sampleSize} 1_{X_i \leq x}

> for all :math:` x \in \Rset`. The test statistic is defined by:

Reviewer 2: ![image](https://github.com/openturns/openturns/assets/31351465/aab812a9-7cc1-4aa7-99f7-c001cc1ecdb4)

```suggestion
for all :math:`x \in \Rset`. The test statistic is defined by:
```

.. math::

    D_{\sampleSize} = \sqrt{\sampleSize} \sup_{x} \left|F_{\sampleSize}\left(x \right) - G\left(x \right)\right|

The empirical value of the test statistic is denoted by :math:`d`, using the realization of
:math:`F_{\sampleSize}` on the sample:

.. math::

    F_{\sampleSize}(x) & = \dfrac{\mbox{number of } x_i \leq x \mbox{ in the sample}}{\sampleSize}

Under the null hypothesis :math:`\mathcal{H}_0 = \{ G = F\}`, the distribution of
the test statistic :math:`D_{\sampleSize}` is
known: algorithms are available to compute the distribution of :math:`D_{\sampleSize}`
both for :math:`\sampleSize`
large (asymptotic distribution: this is the Kolmogorov distribution) or for
:math:`\sampleSize` small (exact distribution). Then we can use that
distribution to apply the test as follows.
We fix a risk :math:`\alpha` (error type I) and we evaluate the associated critical
value :math:`d_\alpha` which is the quantile of order

> :math:`(1-\alpha)` of :math:`D_{\sampleSize}`.

Reviewer 2: ```suggestion
:math:`1-\alpha` of :math:`D_{\sampleSize}`.
```

Then a decision is made, either by comparing the test statistic to the theoretical
threshold :math:`d_\alpha`
(or equivalently
by evaluating the p-value of the sample  defined as
:math:`\Prob{D_{\sampleSize} > d_{\sampleSize}}` and by comparing
it to :math:`\alpha`):

-  if :math:`d_{\sampleSize}>d_{\alpha}` (or equivalently

>    :math:`\Prob{D_{\sampleSize} > d_{\sampleSize}} \leq \alpha`),

Reviewer 2: ```suggestion
   :math:`\Prob{D_{\sampleSize} > d_{\sampleSize}} < \alpha`),
```

Author: idem

    View(graph)

.. topic:: API:

    - See :py:func:`~openturns.FittingTest.Kolmogorov` to compare a sample and a given distribution,
    - See :py:func:`~openturns.FittingTest.BestModelKolmogorov` to select the best model,
    - See :py:func:`~openturns.HypothesisTest.TwoSamplesKolmogorov` to compare two samples
    - See :py:func:`~openturns.FittingTest.Lilliefors` to compare a sample  and a given distribution

>       which parameters habe been estimated from the sample

Reviewer: ```suggestion
      whose parameters have been estimated from the sample
```
